### PR TITLE
test: add comprehensive tests for apted_tree.go

### LINKED ISSUE #236: Add unit tests for tree manipulation functions (apted_tree.go)
## Summary

Add comprehensive unit tests for the tree manipulation functions in `internal/analyzer/apted_tree.go`. These functions are critical for the APTED clone detection algorithm but currently lack test coverage.

### TreeNode Methods
| Function | Description |
|----------|-------------|
| `NewTreeNode(id, label)` | Creates a new tree node |
| `AddChild(child)` | Adds a child node |
| `IsLeaf()` | Checks if node has no children |
| `Size()` | Returns subtree size |
| `SizeWithDepthLimit(maxDepth)` | Returns size with recursion limit |
| `Height()` | Returns subtree height |
| `HeightWithDepthLimit(maxDepth)` | Returns height with recursion limit |
| `String()` | String representation |

### TreeConverter Methods
| Function | Description |
|----------|-------------|
| `NewTreeConverter()` | Creates converter instance |
| `ConvertAST(astNode)` | Converts parser AST to APTED tree |

### Tree Utility Functions
| Function | Description |
|----------|-------------|
| `PostOrderTraversal(root)` | Assigns post-order IDs |
| `ComputeLeftMostLeaves(root)` | Computes left-most leaf descendants |
| `ComputeKeyRoots(root)` | Identifies key roots for path decomposition |
| `PrepareTreeForAPTED(root)` | Prepares tree with all indices |
| `GetNodeByPostOrderID(root, id)` | Finds node by post-order ID |
| `GetSubtreeNodes(root)` | Returns all nodes in subtree |

### Basic TreeNode Tests
```go
func TestNewTreeNode(t *testing.T) {
    node := NewTreeNode(1, "test")
    assert.Equal(t, 1, node.ID)
    assert.Equal(t, "test", node.Label)
    assert.Empty(t, node.Children)
}

func TestAddChild(t *testing.T) {
    parent := NewTreeNode(1, "parent")
    child := NewTreeNode(2, "child")
    parent.AddChild(child)
    
    assert.Len(t, parent.Children, 1)
    assert.Equal(t, parent, child.Parent)
}

func TestAddChild_NilChild(t *testing.T) {
    parent := NewTreeNode(1, "parent")
    parent.AddChild(nil)
    assert.Empty(t, parent.Children)
}

func TestIsLeaf(t *testing.T) {
    leaf := NewTreeNode(1, "leaf")
    assert.True(t, leaf.IsLeaf())
    
    parent := NewTreeNode(1, "parent")
    parent.AddChild(NewTreeNode(2, "child"))
    assert.False(t, parent.IsLeaf())
}
```

### Size and Height Tests
```go
func TestSize_SingleNode(t *testing.T) {
    node := NewTreeNode(1, "root")
    assert.Equal(t, 1, node.Size())
}

func TestSize_TreeWithChildren(t *testing.T) {
    //     root
    //    /    \
    //  child1  child2
    //    |
    //  grandchild
    root := NewTreeNode(1, "root")
    child1 := NewTreeNode(2, "child1")
    child2 := NewTreeNode(3, "child2")
    grandchild := NewTreeNode(4, "grandchild")
    
    root.AddChild(child1)
    root.AddChild(child2)
    child1.AddChild(grandchild)
    
    assert.Equal(t, 4, root.Size())
}

func TestHeight_Leaf(t *testing.T) {
    leaf := NewTreeNode(1, "leaf")
    assert.Equal(t, 0, leaf.Height())
}

func TestHeight_Tree(t *testing.T) {
    // Height 2 tree
    root := NewTreeNode(1, "root")
    child := NewTreeNode(2, "child")
    grandchild := NewTreeNode(3, "grandchild")
    
    root.AddChild(child)
    child.AddChild(grandchild)
    
    assert.Equal(t, 2, root.Height())
}
```

### Post-Order Traversal Tests
```go
func TestPostOrderTraversal(t *testing.T) {
    //     root(2)
    //    /      \
    //  left(0)  right(1)
    root := NewTreeNode(1, "root")
    left := NewTreeNode(2, "left")
    right := NewTreeNode(3, "right")
    
    root.AddChild(left)
    root.AddChild(right)
    
    PostOrderTraversal(root)
    
    assert.Equal(t, 0, left.PostOrderID)
    assert.Equal(t, 1, right.PostOrderID)
    assert.Equal(t, 2, root.PostOrderID)
}
```

## Files to Create

- `internal/analyzer/apted_tree_test.go`

## Reference

Look at similar test patterns in:
- `internal/analyzer/cbo_test.go`
- `internal/analyzer/centroid_grouping_test.go`

## Acceptance Criteria

- [ ] Create `apted_tree_test.go` with tests for all public functions
- [ ] Cover edge cases (nil inputs, empty trees, single nodes)
- [ ] Cover typical use cases (balanced trees, deep trees)
- [ ] All tests pass (`make test`)
- [ ] Test coverage for `apted_tree.go` reaches >80%

## Getting Started

1. Clone the repository
2. Read `internal/analyzer/apted_tree.go` to understand the code
3. Create `internal/analyzer/apted_tree_test.go`
4. Run tests with `go test ./internal/analyzer/... -run TestTreeNode -v`
5. Check coverage with `make coverage`
6. Submit a PR

## Estimated Time

1.5 - 2 hours

---

🏷️ This is a **good first issue** - great for learning the APTED algorithm while contributing!

### PR DESCRIPTION
**Pull request by fuyu28**
Add unit tests for TreeNode methods, TreeConverter, and APTED utility functions in `internal/analyzer/apted_tree.go` to improve coverage and robustness.

  ## Summary
  Add comprehensive unit tests for `internal/analyzer/apted_tree.go` to improve code coverage and reliability for the APTED clone detection algorithm.

  ## Type of Change
  - [ ] Bug fix (non-breaking change)
  - [ ] New feature (non-breaking change)
  - [ ] Breaking change
  - [ ] Documentation update
  - [ ] Performance improvement
  - [ ] Refactoring
  - [x] **Test coverage improvement**

  ## Related Issues
  Fixes #236

  ## Changes
  - Add tests for all TreeNode methods: `NewTreeNode`, `AddChild`, `IsLeaf`, `Size`, `SizeWithDepthLimit`, `Height`, `HeightWithDepthLimit`, `String`
  - Add tests for TreeConverter methods: `NewTreeConverter`, `ConvertAST`
  - Add tests for APTED utility functions: `PostOrderTraversal`, `ComputeLeftMostLeaves`, `ComputeKeyRoots`, `PrepareTreeForAPTED`, `GetNodeByPostOrderID`, `GetSubtreeNodes`
  - Implement table-driven tests for depth limit variants
  - Cover edge cases: nil inputs, negative/zero depth limits, non-existent IDs
  - Cover typical use cases: single nodes, balanced trees, deep trees, complex structures
  - Ensure test isolation: each subtest creates independent tree instances

  ## Testing
  - [x] `make test` passes locally (all 21 tests pass in <1 second)
  - [x] `make lint` passes locally
  - [x] Added tests for new functionality (if applicable) - **494 lines of comprehensive tests**
  - [x] Manual testing completed (if applicable)

  **Test Results:**
  === Test Summary ===
  Total tests: 21
  Passed: 21
  Failed: 0
  Duration: 0.838s

  === Coverage ===
  apted_tree.go: 86.2% (target: >80%)
  All public functions: 100% coverage

  ## Documentation
  - [x] Updated godoc comments - N/A (test file only)
  - [x] Updated README or other docs (if needed) - N/A

  ## Additional Context

  ### Coverage Breakdown
  - **All public functions**: 100% coverage
  - **TreeNode methods**: 100%
  - **TreeConverter**: 100%
  - **Utility functions**: 100%
  - **Overall file coverage**: 86.2%

  ### Test Philosophy Compliance
  Follows guidelines from `docs/TESTING.md`:
  - ✅ Test behavior, not implementation
  - ✅ Fast feedback (<1 second)
  - ✅ Test isolation (independent tree instances)
  - ✅ Clear test names
  - ✅ >80% coverage achieved

  ### Example Test Structure
  ```go
  func TestSizeWithDepthLimit_apted(t *testing.T) {
      tests := []struct {
          name     string
          maxDepth int
          expected int
      }{
          {"negative depth returns 1", -1, 1},
          {"zero depth returns 1", 0, 1},
          {"depth 1 includes direct children", 1, 3},
          // ...
      }
      for _, tt := range tests {
          t.Run(tt.name, func(t *testing.T) {
              // Each subtest creates independent tree
              root := NewTreeNode(1, "root")
              // ...
          })
      }
  }

  This PR completes all acceptance criteria from Issue #236:
  - ✅ Tests for all public functions
  - ✅ Edge cases covered
  - ✅ Typical use cases covered
  - ✅ All tests pass
  - ✅ Coverage >80%

### CODE REVIEW

File: internal/analyzer/apted_tree_test.go [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+			maxDepth: 1,
+			expected: 3, // root + child1 + child2
+		},
+		{
+			name:     "depth 2 include include all nodes",
+			maxDepth: 2,
+			expected: 4, // root + child1 + child2 + grandchild
+		},
+		{
+			name:     "large depth include all nodes",
```
Comment (DaisukeYoda) [inline internal/analyzer/apted_tree_test.go]: "include" → "includes"

File: internal/analyzer/apted_tree_test.go [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+			maxDepth: 1,
+			expected: 1,
+		},
+		{
+			name:     "depth 2 include all nodes",
+			maxDepth: 2,
+			expected: 2,
+		},
+		{
+			name:     "large depth include all nodes",
```
Comment (DaisukeYoda) [inline internal/analyzer/apted_tree_test.go]: line144-154: include should be includes as well

**Review by DaisukeYoda — CHANGES_REQUESTED** [review]
Thanks for working on this PR! The implementation is solid and the tests cover most of the cases. Please fix some minor typos, and it will be ready to merge.
